### PR TITLE
Add Custom Error Page For Prod Mode

### PR DESCRIPTION
**Pull request by gwendolyngoetz**
### Description

This adds a simple custom error page that is displayed when a server error occurs if running in prod mode. Users will no longer see the big scary red error page. The link is a mailto and also sets the default for the body to include the exception id `mailto:civiform-dev@exygy.com?body=[CiviForm Error ID: 81ho6k5dp]`.

_This is mostly a copy of the NotFound error page and doesn't currently include a logo._

image

## Release notes

The title of the pull request will be used as the default release notes description. If more detail is needed to communicate to partners the scope of the PR's changes, use this release notes section.

#### General

- [x] Added the correct label: < feature | bug | dependencies | infrastructure | ignore-for-release >
- [ ] Created tests which fail without the change (if possible)

#### User visible changes

- [ ] Wrote browser tests using the [validateAccessibility](https://sourcegraph.com/github.com/civiform/civiform/-/blob/browser-test/src/support/index.ts?L437:14&subtree=true) method
- [x] Tested on mobile view. See [mobile device mode](https://developer.chrome.com/docs/devtools/device-mode/)
- [x] Manually tested at 200% size
- [x] Manually evaluated tab order
- [x] Add translations

### CODE REVIEW

**Comment by gwendolyngoetz** [comment]
#4822

**Comment by gwendolyngoetz** [comment]
@nb1701 The actual email address is coming from the `support_email_address` in the configuration. If you want I'll open a new PR to change the default email address from civiform-dev to the googlegroups one.

Also this screen only runs when the app is in prod mode.

File: server/conf/messages [hunk 1]
```diff
@@ -549,6 +549,12 @@ error.notFoundTitle=We were unable to find the page you tried to visit
 error.notFoundDescription=Please go back or return to the {0}
 error.notFoundDescriptionLink=homepage
 
+#-------------------------------------------------------#
+# 500 PAGE ERROR - error messages displayed on 500 page #
+#-------------------------------------------------------#
+error.internalServerTitle=An error occurred
```
[the remark this answers is not in the document]
Comment (gwendolyngoetz) [reply server/conf/messages:555]: I don't think that will really mean much to the end user.

File: server/conf/messages [hunk 2]
```diff
@@ -549,6 +549,12 @@ error.notFoundTitle=We were unable to find the page you tried to visit
...
+error.internalServerDescription=Please contact technical support for assistance at %s and include this error id {0}.
```
[the remark this answers is not in the document]
Comment (gwendolyngoetz) [reply server/conf/messages:556]: Not against this, but telling them to include the error is more actionable to me.

File: server/app/views/errors/InternalServerError.java [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+import services.MessageKey;
+import views.BaseHtmlView;
+import views.HtmlBundle;
+import views.LanguageSelector;
+import views.applicant.ApplicantLayout;
+import views.components.LinkElement;
+import views.style.ApplicantStyles;
+import views.style.ErrorStyles;
+
+public final class InternalServerError extends BaseHtmlView {
```
Comment (bion) [inline server/app/views/errors/InternalServerError.java:23]: **nitpick:** please add javadoc for this class

File: server/conf/messages (hunk 2 above)
[the remark this answers is not in the document]
Comment (bion) [reply server/conf/messages:556]: Does %s work? I would guess this needs to be 
```suggestion
error.internalServerDescription=Please contact technical support for assistance at {0} and include this error ID {1}.
```

**Review by bion — CHANGES_REQUESTED** [review]
Please include a screenshot of the error page in the PR description

File: server/app/views/errors/InternalServerError.java (hunk 3 above)
Comment (gwendolyngoetz) [reply server/app/views/errors/InternalServerError.java:23]: Ok, but if I'm adding it here I'm adding it to the 404 page also ;)

Comment (bion) [reply server/app/views/errors/InternalServerError.java:23]: thanks!

**Review by bion — APPROVED** [review]

**Pull request merged**